experimental/sdk/webpubsub-socketio-extension/src/EIO/index.ts

```diff
@@ -19,7 +24,35 @@ import * as engine from "engine.io";
  * TODO: implment BaseServer rather than extends Server
  **/
 export class WebPubSubEioServer extends engine.Server {
+  public webPubSubOptions: WebPubSubExtensionOptions;
+  public webPubSubConnectionManager: WebPubSubConnectionManager;
+
   constructor(options: engine.ServerOptions, webPubSubOptions: WebPubSubExtensionOptions) {
+    debug("create Engine.IO Server with AWPS");
     super(options);
+    this.webPubSubOptions = webPubSubOptions;
+    this.webPubSubConnectionManager = new WebPubSubConnectionManager(this, webPubSubOptions);
+
+    const webPubSubEioMiddleware = this.webPubSubConnectionManager.getEventHandlerEioMiddleware();
+    this.use(webPubSubEioMiddleware);
   }
+
+  protected override createTransport(transportName: string, req: unknown): engine.Transport {
+    debug(`create transport, transportName=${transportName}, force redirect to ${WebPubSubTransport.name}`);
+    return new WebPubSubTransport(req);
+  }
+
+  /**
+   * 
+   * @param req - handshake request
+   * @returns socket id for EIO connection built from handshake request `req`
+   */
+  public override generateId = (req: unknown): string => this.webPubSubConnectionManager.getNextSid();
```

Comment:
getNextSid could return undefined so the type is `string | undefined`?

Reply:
updated